File: medium/Search_in_Rotated_Sorted_Array.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        f = 0
        s = len(nums) - 1
        if nums[f] == target:
                return f
        elif nums[s] == target:
                return s
        while f <= s:
            m = int((s-f)/2) + f
            if nums[m] == target:
                return m
            elif nums[f]<=nums[m]:
                if nums[f]<=target<nums[m]:
                    s = m - 1
                else:
                    f = m + 1
            else:
                if nums[s]>=target>nums[m]:
                    f = m + 1
                else:
                    s = m - 1
        else:
            return -1

File: cpp/medium/Search_in_Rotated_Sorted_Array.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        if (target > nums[0]) {
            int idx = 1;
            int length = nums.size();
            while (idx < length and nums[idx] > nums[idx-1]){
                if (target == nums[idx]){
                    return idx;
                }
                idx++;
            }
        }
        else if (target < nums[0]) {
            int idx = nums.size()-1;
            while (idx > 0 and nums[idx] > nums[idx-1]){
                if (target == nums[idx]){
                    return idx;
                }
                idx--;
            }
            if (target == nums[idx]){
                    return idx;
            }
        }
        else if (target == nums[0]) {
            return 0;
        }
        return -1;
    }
};
